sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -350,6 +351,17 @@ case class Join(
     case UsingJoin(_, _) => false
     case _ => resolvedExceptNatural
   }
+
+  // Ignore hint for canonicalization
+  protected override def doCanonicalize(): LogicalPlan =
+    super.doCanonicalize().asInstanceOf[Join].copy(hint = JoinHint.NONE)
+
+  // Do not include an empty join hint in string description
+  protected override def stringArgs: Iterator[Any] = super.stringArgs.filter { e =>
```

Comment:
how about
```
val hintArg = if (hint.leftHint.isEmpty && hint.rightHint.isEmpty) Nil else Seq(hint)
Seq(left, right, joinType, condition) ++ hintArg
```

Reply:
I was trying to do it in a way that would be "extendable", say, it would work with any future change of the constructor (although we don't expect the constructor of logical operators to change much).